app/models/contact.rb

```diff
@@ -0,0 +1,18 @@
+class Contact < ActiveRecord::Base
+  belongs_to :signature
+
+  validates_presence_of :signature
+  validates_presence_of :address
+  validates_presence_of :phone_number
+
+  validates_length_of :address, maximum: 500
+  validates_length_of :phone_number, maximum: 31
```

Comment:
Sure, I'll can change the style.

I separated the validations into their own lines because I thought it could make reviews easier. As if one alters the _presence_ validation, but not the _length_, then it would be obvious in the diff.

And in general, I've been feeling that they're simpler and map better to the Shoulda tests.

Thoughts?

Reply:
31 chars would be 32 minus the trailing `'\0'`. It's just a _big enough_ length, that's also _not too big_. There's no DB restriction or any better reason for it.

Reply:
Also the specs fail if I remove the presence validation on **signature**:

```
  1) Contact validations should validate that :signature cannot be empty/falsy
     Failure/Error: it { should validate_presence_of(:signature) }

       Expected Contact to validate that :signature cannot be empty/falsy, but
       this could not be proved.
         After setting :signature to ‹nil›, the matcher expected the Contact to
         be invalid, but it was valid instead.
     # ./spec/models/contact_spec.rb:9:in `block (3 levels) in <top (required)>'
     # ./spec/support/database_cleaner.rb:10:in `block (3 levels) in <top (required)>'
     # /home/rogi/.gem/ruby/2.5.7/gems/database_cleaner-1.6.2/lib/database_cleaner/generic/base.rb:16:in `cleaning'
     # /home/rogi/.gem/ruby/2.5.7/gems/database_cleaner-1.6.2/lib/database_cleaner/base.rb:98:in `cleaning'
     # /home/rogi/.gem/ruby/2.5.7/gems/database_cleaner-1.6.2/lib/database_cleaner/configuration.rb:86:in `block (2 levels)
in cleaning'
     # /home/rogi/.gem/ruby/2.5.7/gems/database_cleaner-1.6.2/lib/database_cleaner/configuration.rb:87:in `cleaning'
     # ./spec/support/database_cleaner.rb:9:in `block (2 levels) in <top (required)>'
```

This is what I get on the console:

```
+irb(main):005:0> Contact.new.tap {|x| x.save; pp x.errors.messages }
   (0.5ms)  BEGIN
   (0.8ms)  ROLLBACK
{:address=>["Address must be completed"],
 :phone_number=>["Phone number must be completed"]}
```

Reply:
Ahh, that's because we're not doing `load_defaults "5.2"` in config/application.rb

Reply:
![image](https://user-images.githubusercontent.com/6321/74523420-ec435f80-4f14-11ea-88d4-1099eff61b28.png)

Might be a lot of work to add `load_defaults` 🤔 

Best to leave the validation in place for now I think. 😂